### ISSUE TITLE
Animations only work when presenting controller, not when pushing controller

### ISSUE THREAD

**Issue by AdamMak** [quickbirdstudios/RxCoordinator#21]
I am trying to use a custom animation to push a controller, however the animation does not work when pushing. Only seems to work when presenting. Here is code sample where I have just use the flipping animation taken from the example project.

Here is an example of the code

```
 case .testController:
            var vc = TestController.instantiateController()
            let model = TestModel(coordinator: coordinator)
            
            let animation = Animation(presentationAnimation: CustomAnimations.flippingPresentation,
                                      dismissalAnimation: nil)
            
            vc.bind(to: model)
            return .push(vc, animation: animation)
```

It just seems to ignore the animation object and pushes the controller using standard navigation animation.

However, if I change the push to present

`return .present(vc, animation: animation)`

The animation works as expected.

Is this a bug with the framework? Or am I doing something wrong?

Any help would be much appreciated

Edit: Even in the demo project, it seems the pushing animation example does not work. In the HomeRoute class, there is this example

 ```
case .users:
            let animation = Animation(presentationAnimation: CustomPresentations.flippingPresentation, dismissalAnimation: nil)
            var vc = UsersViewController.instantiateFromNib()
            let viewModel = UsersViewModelImpl(coordinator: coordinator)
            vc.bind(to: viewModel)
            return .push(vc, animation: animation)
```

However it just pushes controller with standard navigation animation.

**Comment by pauljohanneskraft** [comment]
Hey, thanks for pointing out. We already noticed this ourselves and are working on a fix, which will be available in version 1.0.

**Comment by AdMeen89** [comment]
@pauljohanneskraft Hello. Say pls - has it been fixed? At the moment looks like transition animation works partially nice for pushing and do nothing for dismissal(apply default animation). Could you clarify this point?
